From b591553a452b393d25107dbccfa93ca9aef26933 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 17 Feb 2021 00:49:22 +0200
Subject: [PATCH] [SW-29740] gaudi: security: use sram rr for mme3_qm

The MME3_QM's interface clock (LBW) is disabled by default and thus to
access it, it must be enabled. Another option is to protect MME3_QM from
access and thus prevent any "non-clocked" access.
Stretch the SRAM_CFG RR to also cover the MME3_QM, which resides just
before the SRAM_CFG.

Change-Id: Iafc82ee47db6b940d010ca4fd667bd6af143d8a6
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 17b953f..469b3d0 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1167,7 +1167,7 @@
 }
 
 enum gaudi_sec_lbw_rrs {
-	LBW_RR_SRAM_CFG,
+	LBW_RR_MME3_QM_SRAM_CFG,
 	LBW_RR_CPU_TSTAMP,
 	LBW_RR_HBM_GICBGN,
 	LBW_RR_GICEND,
@@ -1194,10 +1194,10 @@
 	 * meaning - not inclusive, so we need to provide it the min address
 	 * below the forbidden one and max address above the forbidden one.
 	 */
-	/* SRAM_CFG up to SIF: RR(LBW_RR_SRAM_CFG) */
-	min = mmSRAM_Y0_X0_BANK_BASE - 1;
+	/* MME3_QM, SRAM_CFG up to SIF: RR(LBW_RR_MME3_QM_SRAM_CFG) */
+	min = mmMME3_QM_BASE - 1;
 	max = mmSIF_RTR_0_BASE;
-	preboot_secure_lbw_rr(LBW_RR_SRAM_CFG, min, max);
+	preboot_secure_lbw_rr(LBW_RR_MME3_QM_SRAM_CFG, min, max);
 
 	/* CPU_TIMESTAMP: RR(LBW_RR_CPU_TSTAMP) */
 	min = mmCPU_TIMESTAMP_BASE - 1;
